### ISSUE TITLE
Add access to editor in schema data validator

### ISSUE THREAD

**Issue by klis87** [ianstormtaylor/slate#2542]
#### Do you want to request a _feature_ or report a _bug_?
Feature

#### What's the current behavior?
Currently you can define many schema rules. I am interested in `data` specifically:
```js
const schema = {
  blocks: {
    image: {
      isVoid: true,
      data: {
        src: v => v && isUrl(v),
      },
    },
  },
}
```
I have a use case that I need to validate a node `data` based on `editor.value.document.data`, which I use to keep some global editor state. Unfortunately in `data` callback in schema rules we dont have access to it.

#### What's the expected behavior?
If we could add `editor` passed as 2nd argument to `data` callback, I could read current `editor.value.document.data` and normalize this node with schema `normalize`. Now I guess I will need to resort to lower level `normalizeNode` with worse performance.

Apologies for potential duplicate #1960 , I guess the answer will be the same, but maybe we could think of a global editor/document data which wouldn't break memoization?

**Comment by klis87** [comment]
`EDIT`: `normalizeNode` also won't work unfortunately, because `normalizeNode` for given nodes isn't called when document.data is updated.

I had a similar problem in `renderNode`, which I solved by using react context. For this issue though context won't help.

**Comment by ianstormtaylor** [comment]
Hey @klis87, I understand your need, but like you said I've already addressed this in #1960. In the current architecture for how normalizing works this isn't really possible.
